### ISSUE TITLE
Invoke-ExternalCommand returns too early after waiting for an uninstall process

### ISSUE THREAD

**Issue by Harakku** [lukesampson/scoop#3771]
Sometimes Invoke-ExternalCommand returns too early which can cause errors if Scoop tries to remove an installation directory while an uninstaller is still running.

During my testing, Scoop ended up throwing an error about 50% of the times. I was testing [this manifest](https://github.com/Harakku/harakkus-scoop-bucket/blob/master/bucket/leagueoflegends.json) and here's the uninstaller script I used:
`"script": "Invoke-ExternalCommand \"$dir\\Riot Games\\League of Legends\\Uninstall League of Legends.exe\" -ArgumentList \"--mode unattended\" -RunAs"`

![invoke-externalcommand](https://user-images.githubusercontent.com/49722812/69589714-ace03380-0ff5-11ea-982e-7e6e76389733.gif)

On the other hand, Start-Process returns every time precisely when an uninstaller closes, at least based on my testing. That is noteworthy since both commands seem to have been implemented using the same System.Diagnostics.Process class. The script I used:
`"script": "Start-Process \"$dir\\Riot Games\\League of Legends\\Uninstall League of Legends.exe\" -ArgumentList \"--mode unattended\" -Verb RunAs -Wait"`

![start-process](https://user-images.githubusercontent.com/49722812/69589719-b073ba80-0ff5-11ea-9a1d-5d11944d0619.gif)

I tried both system-installed Powershell 5 and pre-release of Powershell 7. I also tried to run the commands from both an administrator shell and a user shell with a UAC popup. There was no difference. The problem also exists when taking only the [Invoke-ExternalCommand function](https://github.com/lukesampson/scoop/blob/5643818e1b4831f39821549fb8d97c6a16b43c3f/lib/core.ps1#L375) to a new file and running it separated from Scoop.

I have no idea why the Scoop function and [Start-Process](https://github.com/PowerShell/PowerShell/blob/d58a82ad19fbfad81e85778c8b08cb1b28f58fce/src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs#L1605) work differently but one solution could be to rework the Scoop function into using Start-Process behind the scenes instead of System.Diagnostics.Process class directly.

**Comment by niheaven** [comment]
We use `System.Diagnostics.Process` mainly because of log output for `-Verb RunAs`.

Do you have more examples for this early exit?

**Comment by Harakku** [comment]
I tried but didn't manage to make a simplified version of the bug and I've only seen it happen with this one uninstaller.

**Comment by Harakku** [comment]
I wanted  to see if logging would reveal something so I replaced the "-RunAs" with "-Log C:\\log.txt" and ran powershell as admin. I ran install/uninstall over ten times and there were no errors. Interesting.jpg. Then I removed the -Log parameter and there were errors again. It seems that -Log is adding a little bit of delay at the end of the function which makes everything run smoothly.

At first, I was pretty sure that the inherent problem wouldn't be in the uninstaller because Start-Process was managing it just fine. Now I am not sure if the reason for the success of Start-Process is just possibly more delay at the end.

Since I'm not sure if I should point finger to the uninstaller or Scoop, I'll close this for now and come back if more info is revealed.
